src/test/scala/ServiceSpec.scala

```diff
@@ -0,0 +1,127 @@
+import cats.implicits._
+import org.scalatest._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+
+class ServiceSpec extends FlatSpec with Matchers {
+
+  "UserRepositoryId" should "save and retrieve User" in {
+    val username = "username"
+    val wrongUsername = "bla-bla-bla"
+    val wrongId = 999
+    val userRepository = new UserRepositoryId()
+    val savedUser = userRepository.registerUser(username)
+    savedUser.username shouldEqual username
+    userRepository.getById(savedUser.id) shouldEqual Option(savedUser)
+    userRepository.getByUsername(username) shouldEqual Option(savedUser)
+    userRepository.getById(wrongId) shouldEqual Option.empty
+    userRepository.getByUsername(wrongUsername) shouldEqual Option.empty
+  }
+
+  "UserRepositoryFuture" should "save and retrieve User" in {
+    val username = "username"
+    val wrongUsername = "bla-bla-bla"
+    val wrongId = 999
+    val userRepository = new UserRepositoryFuture()
+    val savedUser = await(userRepository.registerUser(username))
+    savedUser.username shouldEqual username
+    await(userRepository.getById(savedUser.id)) shouldEqual Option(savedUser)
```

Comment:
consider please also http://www.scalatest.org/user_guide/async_testing instead of blocking